sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -143,13 +148,15 @@ class RegexParser(pattern: String) {
         parseEscapedCharacter()
       case '\u0000' =>
         throw new RegexUnsupportedException(
-          "cuDF does not support null characters in regular expressions", Some(pos))
+          "cuDF does not support null characters in regular expressions", Some(pos-1))
       case '*' | '+' | '?' =>
         throw new RegexUnsupportedException(
-          "base expression cannot start with quantifier", Some(pos))
+          "base expression cannot start with quantifier", Some(pos-1))
```

Comment:
nit: Error messages are currently inconsistent with respect to the capitalization of the first word. I suggest starting them all with upper-case with the exception of messages starting with "cuDF ..."

```suggestion
          "Base expression cannot start with quantifier", Some(pos-1))
```